pkg/cache/cluster.go

```diff
@@ -1047,6 +1071,56 @@ func (c *clusterCache) onNodeRemoved(key kube.ResourceKey) {
 	}
 }
 
+func (c *clusterCache) updateChildrenByParentMap(res *Resource) {
+	if res == nil || !c.showClusterResourceChildren {
+		return
+	}
+	childKey := res.ResourceKey()
+	if _, ok := c.childrenByParent[childKey]; !ok {
+		c.childrenByParent[childKey] = []kube.ResourceKey{}
+	}
+	for _, parent := range res.OwnerRefs {
+		parentGvk := schema.FromAPIVersionAndKind(parent.APIVersion, parent.Kind)
+		var namespace string
+		if isNamespaced, _ := c.IsNamespaced(parentGvk.GroupKind()); isNamespaced {
+			namespace = res.Ref.Namespace
+		}
+		parentKey := kube.NewResourceKey(parentGvk.Group, parentGvk.Kind, namespace, parent.Name)
+		childRefs, ok := c.childrenByParent[parentKey]
+		if !ok {
+			c.childrenByParent[parentKey] = []kube.ResourceKey{}
+		}
+		exists := false
+		for _, cr := range childRefs {
+			if cr == childKey {
+				exists = true
+			}
+		}
+		if !exists {
+			childRefs = append(childRefs, childKey)
+		}
+		c.childrenByParent[parentKey] = childRefs
+	}
+}
+
+func (c *clusterCache) removeFromChildrenByParentMap(key kube.ResourceKey) {
+	if !c.showClusterResourceChildren {
+		return
+	}
+
+	delete(c.childrenByParent, key)
+
+	for k, childRefs := range c.childrenByParent {
+		for i, childRef := range childRefs {
+			childRefKey := kube.NewResourceKey(childRef.Group, childRef.Kind, childRef.Namespace, childRef.Name)
+			if childRefKey == key {
+				childRefs[i] = childRefs[len(childRefs)-1]
+				c.childrenByParent[k] = childRefs[:len(childRefs)-1]
```

Comment:
This is clever, and I hate how many times I had to read it to understand it. 😆

Reply:
Added comments to improve readability.